### PR TITLE
Make config items of Function type in SWAGGER_UI_CONFIG work

### LINKED ISSUE #381: Cannot provide requestInterceptor option in SWAGGER_UI_CONFIG
I tried to provide the "requestInterceptor" option to Swagger UI via SWAGGER_UI_CONFIG, but it did not work. The basic problem seems to be that apiflask assumes that all such options should be strings, but this is not the case for this option. If only some of the options are supported, this should be documented.

To reproduce, take your own example program and add the lines
```
app.config['SWAGGER_UI_CONFIG'] = {
   'requestInterceptor': "(req) => { console.log('intercepted!'); return req; }"
}
```
Navigate to the docs page and it fails with the error "r.requestInterceptor is not a function /openapi.json"
But there does not seem to be any way to make it into a function in my Python code. I have done something similar in C# with Swashbuckle, where it works fine to provide it as a string in this way.

In ui_templates.py in apiflask are the lines

 ```
var userConfig = {{ config.SWAGGER_UI_CONFIG | tojson }}
    for (var attr in userConfig) {
      baseConfig[attr] = userConfig[attr]
    }
```
which seems to just assume that all options are meant to be strings. If I change this to `eval(userConfig[attr])` then it works (but would then not work for options that are meant to be strings rather than javascript code, so some kind of conditional logic would be needed. )

Environment:

- Python 3.10.6
- Flask version 2.1.3
- APIFlask version: 1.1.3

### PR DESCRIPTION
**Pull request by tkzt**
fixes #381

Checklist:

- [ ] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [x] Add or update relevant docs, in the `docs` folder and in code docstring.
- [x] Add an entry in `CHANGES.md` summarizing the change and linking to the issue.
- [x] Add `*Version changed*` or `*Version added*` note in any relevant docs and docstring.
- [x] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -1004,6 +1004,17 @@ app.config['SWAGGER_UI_CONFIG'] = {
 }
 ```
 
+!!! tip
+
+    Configurations of `Function` type can be provided through strings after  [version 1.2.2](/changelog/#version-122).
```
Comment (greyli) [inline docs/configuration.md:1009]:
```suggestion
    Configurations of `Function` type can be provided through strings after [version 1.2.2](/changelog/#version-122).
```

**Review by greyli — COMMENTED** [review]
Could you add some tests to validate this feature?

**Comment by tkzt** [comment]
> Could you add some tests to validate this feature?

Actually I've tried to add a test case to validate, but since all swagger-ui configurations will only take effect after HTML rendering, finally I found there is no way to check whether one certain configuration work as expected without a HTML rendering framework (such as requests-html, selenium). 

And frameworks with HTML rendering capability are always built on top of a chromium core, which means we need to download a 100+mb file first before running test cases. So I think maybe more discussions on whether adding a HTML rendering framework to dependencies are needed.

**Comment by greyli** [comment]
Oh, I see. Thanks for the detailed explanation. I'll just merge this (tested locally & improved the code style).

**Pull request merged**